provider/resources.go

```diff
@@ -90,11 +94,55 @@ func Provider() tfbridge.ProviderInfo {
 				}},
 			"confluent_kafka_cluster_config": {Tok: tfbridge.MakeResource(mainPkg, mainMod, "KafkaClusterConfig")},
 			"confluent_kafka_mirror_topic":   {Tok: tfbridge.MakeResource(mainPkg, mainMod, "KafkaMirrorTopic")},
-			"confluent_kafka_topic":          {Tok: tfbridge.MakeResource(mainPkg, mainMod, "KafkaTopic")},
-			"confluent_network":              {Tok: tfbridge.MakeResource(mainPkg, mainMod, "Network")},
-			"confluent_peering":              {Tok: tfbridge.MakeResource(mainPkg, mainMod, "Peering")},
-			"confluent_private_link_access":  {Tok: tfbridge.MakeResource(mainPkg, mainMod, "PrivateLinkAccess")},
-			"confluent_role_binding":         {Tok: tfbridge.MakeResource(mainPkg, mainMod, "RoleBinding")},
+			"confluent_kafka_topic": {
+				Tok: tfbridge.MakeResource(mainPkg, mainMod, "KafkaTopic"),
+				PreCheckCallback: func(
+					ctx context.Context, config resource.PropertyMap, _ resource.PropertyMap,
+				) (resource.PropertyMap, error) {
+					const (
+						httpEndpoint = "httpEndpoint"
+						restEndpoint = "restEndpoint"
+					)
+					endpoint, ok := config[httpEndpoint]
+					if !ok {
+						// The optional key is not present, so we can return the config as is.
+						return config, nil
+					}
+
+					// The user has set both httpEndpoint and restEndpoint. These are semantically
+					// the same, so it is forbidden to set both.
+					if _, ok := config[restEndpoint]; ok {
+						return config, fmt.Errorf(`Cannot specify both "httpEndpoint" and "restEndpoint"` +
```

Comment:
super nitty, feel fee to ignore: since we're defining these as `const`s, let's use them here too.
```suggestion
						return config, fmt.Errorf(`Cannot specify both %s and %s` + ...`, httpEndpoint, restEndpoint
```